[12.0][MIG] beesdoo_shift, beesdoo_website_shift, beesdoo_worker_status

elouan-lb opened the pull request:
Migration of `beesdoo_shift` and `beesdoo_website_shift` from version 9 to version 12.

## beesdoo_shift

### AttendanceSheet
- Method `_generate_missing_attendance_sheets()` is added, allowing generation of past missing sheets.

Commit:
[FIX] b_shift : fix cron jobs definition to fit version 12.0 requirements

Commit:
[FIX] beesdoo_shift : Error when saving a temporary exempt

Add missing sequence in database.

Commit:
[FIX] beesdoo_shift : Error when saving a temporary exempt

Add beesdoo_shift_temporary_exemption_id_seq.

Commit:
[FIX] b_shift : fix datetime issues coming from migration to 12.0

Commit:
[ADD] b_shift : attendance sheet models

Commit:
[ADD] b_shift : attendance_sheet_id to TaskTemplate

Commit:
[ADD] b_shift : attendance sheet views

Commit:
[ADD] b_shift : wizard for attendance sheet generation

Commit:
[ADD] b_shift : attendance sheet access rules

Commit:
[ADD] b_shift : attendance sheet validation wizard

Commit:
[ADD] b_shift : attendance sheet notifications count

Commit:
[FIX] b_shift : warning messages

Commit:
[ADD] b_shift : attendance sheet list items coloration

Commit:
[FIX] b_shift : replacement workers must be regular and not unsuscribed

Commit:
[ADD] b_shift : constraint on worker, cannot replace himself

Commit:
[FIX] b_shift : attendance sheet replacement worker field name

Commit:
[IMP] b_shift : attendance sheet views set no_open options

Commit:
[IMP] b_shift : attendance sheet display date with ISO format

Commit:
[ADD] b_shift : attendance shift validation barcode field

Commit:
[ADD] b_shift : default task type configuration

Commit:
[IMP] b_shift : constraint on validated attendance sheets

Commit:
[ADD] b_shift : attendance sheet "mark as read" smart button

Commit:
[ADD] b_shift : attendance sheet archived attribute

Commit:
[ADD] b_shift : demo datas

Commit:
[ADD] b_shift : attendance sheets barcode scanner support

Commit:
[REM] b_shift : attendance sheet cancelled status

Commit:
[FIX] b_shift : attendance sheet shifts worker not required

Commit:
[IMP] b_shift : attendance sheets linting and menu sequences

Commit:
[FIX] b_shift : comment attendance sheet unique worker constraint

Commit:
[IMP] b_shift : change and add access rules

Commit:
[IMP] b_shift : attendance sheets code cleaning

Commit:
[FIX] b_shift : constrain_unique_worker on attendance sheets

Commit:
[FIX] b_shift : code cleaning

Commit:
[IMP] b_shift : shift start_time, end_time required

Commit:
[IMP] b_shift : attendance sheet validation exceptions raised

Commit:
[ADD] b_shift : barcode scan to attendance sheet validation

Commit:
[FIX] b_shift : attendance sheet compensation_nb onchange issues

The @api.onchange, modifying compensation_nb according to working mode,
led to strange behavior and lack of consistency.

compensation_nb has consequently moved in Shift Stage.

Commit:
[FIX] b_shift : barcode scan to attendance sheet validation

Commit:
[REM] beesdoo_shift_assets

Commit:
[FIX] b_shift : attendance sheet scanning methods

Commit:
[ADD] b_shift : constraint on irregular_start_date

Commit:
[IMP] b_shift : attendance sheet form view

Commit:
[REM] b_shift : expected_worker_nb and added_worker_nb from attendance sheet

They are already automatically displayed on the form view.

Commit:
[IMP] b_shift : attendance sheet validation button hidden when validated

Commit:
[ADD] b_shift : attendance sheet possibility to translate terms

Commit:
[FIX] b_shift : date format %%Y-%%m-%%d

Commit:
[ADD] b_shift : attendance sheet CRON for generation

Commit:
[FIX] b_shift : code cleaning

Commit:
[REM] b_shift : generate_attendance_sheet views

Commit:
[ADD] b_shift : attendance sheet default super cooperator

Commit:
[ADD] b_shift : attendance sheet search view

Commit:
[FIX] b_shift : computed methods not applying to recordset

Commit:
[ADD] b_shift : attendance sheet time slot

Commit:
[ADD] b_shift : demo datas

Commit:
Revert "[ADD] b_shift : demo datas"

This reverts commit 6c0d0a0d8c2952628c4d26d0acc4d115f042e118.

Commit:
[FIX] b_shift : stored field in abstract class

Commit:
[FIX] b_shift : attendance sheet, validated by

Commit:
[FIX] b_shift : attendance sheet annotation validation

Commit:
[FIX] b_shift : wrong default regular task type

Commit:
[REF] b_shift : attendance sheet code cleaning

Commit:
[ADD] b_shift : attendance sheet automatic emails

Commit:
[IMP] b_shift : change attendance sheets shifts order

Commit:
[ADD] b_w_shift : prevent subscription 20 min before start

Commit:
[IMP] b_shift : change configuration menu order

Commit:
[ADD] b_shift : attendance sheet tests setUp

Commit:
[FIX] b_shift : search on equal dates

Commit:
[ADD] b_shift : attendance sheets test on creation

Commit:
[ADD] b_shift : attendance sheet test on default task type

Commit:
[ADD] b_shift : member cards demo datas

Commit:
[ADD] b_shift : exception raised when replaced worker try to subscribe

Commit:
[ADD] b_shift : attendance sheet unit test on barcode scanner

Commit:
[IMP] b_shift : no exception raised when scanning a worker twice

Commit:
[IMP] b_shift : cancelled shift not added to attendance sheets

Commit:
[IMP] b_shift : attendance sheet max worker no calculation

Now based on created shifts instead of task templates

Commit:
Revert "[ADD] b_shift : attendance sheet default super cooperator"

This reverts commit 605ac1853f6ab0497bb0a6bf566c66655056269d.

Commit:
[REM] b_shift : attendance sheet shift 'cancelled' status

Commit:
[REF] b_shift : code cleaning

Commit:
[IMP] b_shift : cooperative admin can validate sheet with their cards

Commit:
[FIX] b_shift : cron for non-validated sheets

Commit:
[UPD] b_shift : attendance sheet name splitted in day and time_slot on view

Commit:
[IMP] b_shift : attendance annotation pre-filled on wizard

Commit:
[REM] b_shift : Shift Attendance view

Commit:
[IMP] b_shift : group_shift_attendance can accessShift menu

Commit:
[IMP] b_shift : attendance sheet generation time configurable

Generation CRON is running every minute and check for sheets
in the configurable time interval. Workers can't subscribe
to a shift on intranet if its start time is in this interval.

Commit:
[IMP] b_shift : attendance sheet generation time configurable

Generation CRON is running every minute and check for sheets
in the configurable time interval. Workers can't subscribe
to a shift on intranet if its start time is in this interval.

Commit:
[ADD] b_shift : attendance sheet, is_compensation field for expected shifts

Commit:
[ADD] b_shift : week name on attendance sheet

Commit:
[WIP] b_shift : give access to backend to attendance sheets generic group

Commit:
[ADD] b_shift : attendance sheet compensation number

Changes from shift status update are made as well.

Commit:
[FIX] b_shift : time on added shift

Commit:
[ADD] b_shift : weekly summary email

Commit:
[FIX] b_shift : next_countdown computation

When the next_countdown is the same day than the current one,
it should be the one displayed (and not the one 28 days later).

Commit:
[REF] b_shift : code cleaning

Commit:
[IMP] b_shift : automatically created shifts names

Commit:
[CHG] b_shift : checks on attendance sheet before validation

Commit:
[REF] b_shift : views order and definition

Commit:
[ADD] b_shift : replacement worker on shift kanban view

Commit:
[ADD] b_shift : optional validation by card scan

Commit:
[ADD] b_shift : constrains on future attendance sheets validation

Commit:
[FIX] b_shift : automatic unsubscribe from next shifts

Doesn't unsubscribe from present shift anymore
when worker is changing to "unsubscribed" status.

You cannot change future shifts status anymore.

Commit:
[FIX] b_shift : automatic unsubscribe from next shifts

Doesn't unsubscribe from present shift anymore
when worker is changing to "unsubscribed" status.

You cannot change future shifts status anymore.

Commit:
[ADD] b_shift : day_abbreviation for attendance sheets

Commit:
[FIX] b_shift : cron no_update='1'

Commit:
[FIX] b_shift : attendance sheet 'is_annotated'

Commit:
[IMP] b_shift : attendance sheet validation wizard

Temporary changes can be saved.
Warnings added and modified.

Commit:
[FIX] b_shift : sheet generation cron overloading ram

Commit:
[REF] b_shift : code cleaning

Commit:
[ADD] b_shift : _order for attendance sheet shifts

Commit:
[REF] b_shift : code cleaning

Commit:
[ADD] b_shift : group_by on worker_no_feedback

Commit:
[CHG] b_shift : shift status

To give better understanding of compensations given by each status.

'stage_id' is replaced by a selection field state to ease
domain restrictions.

Commit:
[CHG] b_shift : shift status

To give better understanding of compensations given by each status.

'stage_id' is replaced by a selection field state to ease
domain restrictions.

Commit:
[ADD] b_shift : demo datas for task templates

Commit:
[FIX] b_shift : supercoop name not displayed on sheets

Commit:
Revert "[FIX] b_shift : automatic unsubscribe from next shifts"

This reverts commit 796753a5edaea9883501b54920083a3693b083bf.

Commit:
[UPD] b_shift : demos datas & unite tests

Commit:
[CHG] b_shift : attendance sheets terminology

Commit:
[FIX] b_shift : search on datetime with '='

Commit:
[REF] b_shift : code cleaning

Commit:
[FIX] b_shift : attendance sheet _compute_is_annotated

Commit:
[REF] b_shift : code cleaning

Commit:
[IMP] b_shift : attendance sheet validation wizard

Commit:
[ADD] b_shift : tests for cooperative status

Commit:
[ADD] b_shift : authors names

Commit:
[MOV] b_shift : demo datas for users in b_base

Commit:
[CHG] b_shift : _cron_non_validated_sheets inactive by default

Commit:
[CHG] b_shift : info_session to date instead of datetime

Commit:
[MIG] b_shift : code cleaning for 12.0

Commit:
[ADD] b_shift : checks for holidays and exemption

Commit:
[MIG] b_shift: migration to 10.0

Commit:
[MIG] b_shift : migration to 11.0

Commit:
[MIG] b_shift : migration to 12.0

Commit:
[MIG] b_shift : 'ir.needaction_mixin' removed from 11.0

Commit:
[ADD] b_shift : traductions

Commit:
[FIX] b_shift : email for non-attended shifts

Commit:
[FIX] b_shift : created shifts not updating counters

If one shift is directly created as "Attended",
workers' shifts counters are not updated.
They are only updated on state transitions.

Commit:
[UPD] b_shift : unite tests

Commit:
[MIG] b_shift : datetimes aren't strings anymore

Commit:
[MIG] b_shift : fix views

Commit:
[FIX] b_shift : demo datas for member cards

Commit:
[MIG] b_shift : fix security groups

remytms committed:
[MIG] beesdoo_website_shift: Migrate to 12.0

Commit:
[MIG] b_shift : change settings to fit 12.0 requirements

Commit:
[MIG] b_shift : change cron definitions to fit 12.0 requirements

Commit:
[MIG] b_shift : fix validation wizard to fit 12.0 requirements

Commit:
[MIG] b_shift : fix views to fit 12.0 requirements

Commit:
[MIG] b_shift : i18n for 12.0

Commit:
[REF] b_shift : code cleaning

Commit:
[FIX] b_shift : error messages

Commit:
[UPD] b_shift : traductions

Commit:
[FIX] b_shift : security groups for coop admin

Commit:
[ADD] b_shift : constrains on exempted workers

Commit:
[REF] b_shift : remove state computation for attendance shifts

replacement_worker_id -> replaced_id

Commit:
[REF] b_shift : demos

Commit:
[ADD] b_shift : domain on attendance shifts

Commit:
[FIX] b_shift : automatic error message

For regular non-attended not doing
their compensation shift.

Commit:
[UPD] b_shift : security groups

Add a new groupe only for attendance
sheets validation.

Commit:
[ADD] b_shift : option for worker_nb feedback

Commit:
[UPD] b_shift : attendance sheet form view

Commit:
[FIX] b_shift : attendance sheet generation

When several same shifts exists

Commit:
fixup! [FIX] b_shift : email_template_non_attendance

Commit:
[UPD] b_shift : translations

Commit:
[UPD] b_shift : attendance sheets view

To ease scanner use.

StCyr commented:
Hello @elouan-lb 

Trying to use your branch here result in:

```
2020-02-03 08:11:20,354 15476 ERROR odoo12 odoo.sql_db: bad query:
                INSERT INTO ir_model_data (module, name, model, res_id, noupdate, date_init, date_update)
                VALUES ('beesdoo_shift', 'create_beesdoo_shift_shift', 'ir.model.access', 491, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'read_beesdoo_shift_sheet_shift', 'ir.model.access', 492, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'create_beesdoo_shift_sheet_expected', 'ir.model.access', 493, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_sheet_added', 'ir.model.access', 494, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'create_beesdoo_shift_sheet', 'ir.model.access', 495, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'sheet_access_beesdoo_shift_template', 'ir.model.access', 496, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'sheet_access_beesdoo_shift_type', 'ir.model.access', 497, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'access_beesdoo_shift_daynumber', 'ir.model.access', 498, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'read_beesdoo_shift_planning', 'ir.model.access', 499, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'access_beesdoo_shift_template', 'ir.model.access', 500, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'write_beesdoo_shift_shift', 'ir.model.access', 501, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_shift', 'ir.model.access', 502, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_type', 'ir.model.access', 503, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_daynumber', 'ir.model.access', 504, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_planning', 'ir.model.access', 505, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_template', 'ir.model.access', 506, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_sheet_shift', 'ir.model.access', 507, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_sheet_expected', 'ir.model.access', 508, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_cooperative_status', 'ir.model.access', 509, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_cooperative_exempt_reason', 'ir.model.access', 510, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'read_beesdoo_shift_journal', 'ir.model.access', 511, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'access_cooperative_status', 'ir.model.access', 512, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'read_cooperative_exempt_reason', 'ir.model.access', 513, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'read_cooperative_status_history', 'ir.model.access', 514, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'access_beesdoo_shift_type', 'ir.model.access', 515, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'access_beesdoo_shift_type', 'ir.model.access', 516, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'access_beesdoo_shift_template', 'ir.model.access', 517, false, now() at time zone 'UTC', now() at time zone 'UTC')
                ON CONFLICT (module, name)
                DO UPDATE SET date_update=(now() at time zone 'UTC')

ERROR: ON CONFLICT DO UPDATE command cannot affect row a second time
HINT:  Ensure that no rows proposed for insertion within the same command have duplicate constrained values.

2020-02-03 08:11:20,355 15476 ERROR odoo12 odoo.addons.base.models.ir_model: Failed to insert ir_model_data
('beesdoo_shift', 'create_beesdoo_shift_shift', 'ir.model.access', 491, False)
('beesdoo_shift', 'read_beesdoo_shift_sheet_shift', 'ir.model.access', 492, False)
('beesdoo_shift', 'create_beesdoo_shift_sheet_expected', 'ir.model.access', 493, False)
('beesdoo_shift', 'manage_beesdoo_shift_sheet_added', 'ir.model.access', 494, False)
('beesdoo_shift', 'create_beesdoo_shift_sheet', 'ir.model.access', 495, False)
('beesdoo_shift', 'sheet_access_beesdoo_shift_template', 'ir.model.access', 496, False)
('beesdoo_shift', 'sheet_access_beesdoo_shift_type', 'ir.model.access', 497, False)
('beesdoo_shift', 'access_beesdoo_shift_daynumber', 'ir.model.access', 498, False)
('beesdoo_shift', 'read_beesdoo_shift_planning', 'ir.model.access', 499, False)
('beesdoo_shift', 'access_beesdoo_shift_template', 'ir.model.access', 500, False)
('beesdoo_shift', 'write_beesdoo_shift_shift', 'ir.model.access', 501, False)
('beesdoo_shift', 'manage_beesdoo_shift_shift', 'ir.model.access', 502, False)
('beesdoo_shift', 'manage_beesdoo_shift_type', 'ir.model.access', 503, False)
('beesdoo_shift', 'manage_beesdoo_shift_daynumber', 'ir.model.access', 504, False)
('beesdoo_shift', 'manage_beesdoo_shift_planning', 'ir.model.access', 505, False)
('beesdoo_shift', 'manage_beesdoo_shift_template', 'ir.model.access', 506, False)
('beesdoo_shift', 'manage_beesdoo_shift_sheet_shift', 'ir.model.access', 507, False)
('beesdoo_shift', 'manage_beesdoo_shift_sheet_expected', 'ir.model.access', 508, False)
('beesdoo_shift', 'manage_cooperative_status', 'ir.model.access', 509, False)
('beesdoo_shift', 'manage_cooperative_exempt_reason', 'ir.model.access', 510, False)
('beesdoo_shift', 'read_beesdoo_shift_journal', 'ir.model.access', 511, False)
('beesdoo_shift', 'access_cooperative_status', 'ir.model.access', 512, False)
('beesdoo_shift', 'read_cooperative_exempt_reason', 'ir.model.access', 513, False)
('beesdoo_shift', 'read_cooperative_status_history', 'ir.model.access', 514, False)
('beesdoo_shift', 'access_beesdoo_shift_type', 'ir.model.access', 515, False)
('beesdoo_shift', 'access_beesdoo_shift_type', 'ir.model.access', 516, False)
('beesdoo_shift', 'access_beesdoo_shift_template', 'ir.model.access', 517, False)
```

procedure followed:
1. dropdb odoo12
2. createdb odoo12
3. odoo/odoo-bin -c ./odoo.conf -d odoo12 -i beesdoo_base
4. odoo/odoo-bin -c ./odoo.conf -d odoo12 -i beesdoo_shift

tfrancoi commented:
C'est le fichier ir.model.access qui a des dupliquer .
il faut les retirer comme dans ce patch
https://github.com/beescoop/Obeesdoo/commit/ba090eb3b2a337a416c4fe684e8edbd555daefb3#diff-644e5bdc4b769f4f8799c114f4c10e70R27

Le lun. 3 févr. 2020 à 09:13, Cyrille Bollu <notifications@github.com> a
écrit :

> Hello @elouan-lb <https://github.com/elouan-lb>
>
> Trying to use your branch here result in:
>
> 2020-02-03 08:11:20,354 15476 ERROR odoo12 odoo.sql_db: bad query:
>                 INSERT INTO ir_model_data (module, name, model, res_id, noupdate, date_init, date_update)
>                 VALUES ('beesdoo_shift', 'create_beesdoo_shift_shift', 'ir.model.access', 491, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'read_beesdoo_shift_sheet_shift', 'ir.model.access', 492, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'create_beesdoo_shift_sheet_expected', 'ir.model.access', 493, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_sheet_added', 'ir.model.access', 494, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'create_beesdoo_shift_sheet', 'ir.model.access', 495, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'sheet_access_beesdoo_shift_template', 'ir.model.access', 496, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'sheet_access_beesdoo_shift_type', 'ir.model.access', 497, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'access_beesdoo_shift_daynumber', 'ir.model.access', 498, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'read_beesdoo_shift_planning', 'ir.model.access', 499, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'access_beesdoo_shift_template', 'ir.model.access', 500, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'write_beesdoo_shift_shift', 'ir.model.access', 501, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_shift', 'ir.model.access', 502, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_type', 'ir.model.access', 503, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_daynumber', 'ir.model.access', 504, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_planning', 'ir.model.access', 505, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_template', 'ir.model.access', 506, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_sheet_shift', 'ir.model.access', 507, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_beesdoo_shift_sheet_expected', 'ir.model.access', 508, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_cooperative_status', 'ir.model.access', 509, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'manage_cooperative_exempt_reason', 'ir.model.access', 510, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'read_beesdoo_shift_journal', 'ir.model.access', 511, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'access_cooperative_status', 'ir.model.access', 512, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'read_cooperative_exempt_reason', 'ir.model.access', 513, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'read_cooperative_status_history', 'ir.model.access', 514, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'access_beesdoo_shift_type', 'ir.model.access', 515, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'access_beesdoo_shift_type', 'ir.model.access', 516, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('beesdoo_shift', 'access_beesdoo_shift_template', 'ir.model.access', 517, false, now() at time zone 'UTC', now() at time zone 'UTC')
>                 ON CONFLICT (module, name)
>                 DO UPDATE SET date_update=(now() at time zone 'UTC')
>
> ERROR: ON CONFLICT DO UPDATE command cannot affect row a second time
> HINT:  Ensure that no rows proposed for insertion within the same command have duplicate constrained values.
>
> 2020-02-03 08:11:20,355 15476 ERROR odoo12 odoo.addons.base.models.ir_model: Failed to insert ir_model_data
> ('beesdoo_shift', 'create_beesdoo_shift_shift', 'ir.model.access', 491, False)
> ('beesdoo_shift', 'read_beesdoo_shift_sheet_shift', 'ir.model.access', 492, False)
> ('beesdoo_shift', 'create_beesdoo_shift_sheet_expected', 'ir.model.access', 493, False)
> ('beesdoo_shift', 'manage_beesdoo_shift_sheet_added', 'ir.model.access', 494, False)
> ('beesdoo_shift', 'create_beesdoo_shift_sheet', 'ir.model.access', 495, False)
> ('beesdoo_shift', 'sheet_access_beesdoo_shift_template', 'ir.model.access', 496, False)
> ('beesdoo_shift', 'sheet_access_beesdoo_shift_type', 'ir.model.access', 497, False)
> ('beesdoo_shift', 'access_beesdoo_shift_daynumber', 'ir.model.access', 498, False)
> ('beesdoo_shift', 'read_beesdoo_shift_planning', 'ir.model.access', 499, False)
> ('beesdoo_shift', 'access_beesdoo_shift_template', 'ir.model.access', 500, False)
> ('beesdoo_shift', 'write_beesdoo_shift_shift', 'ir.model.access', 501, False)
> ('beesdoo_shift', 'manage_beesdoo_shift_shift', 'ir.model.access', 502, False)
> ('beesdoo_shift', 'manage_beesdoo_shift_type', 'ir.model.access', 503, False)
> ('beesdoo_shift', 'manage_beesdoo_shift_daynumber', 'ir.model.access', 504, False)
> ('beesdoo_shift', 'manage_beesdoo_shift_planning', 'ir.model.access', 505, False)
> ('beesdoo_shift', 'manage_beesdoo_shift_template', 'ir.model.access', 506, False)
> ('beesdoo_shift', 'manage_beesdoo_shift_sheet_shift', 'ir.model.access', 507, False)
> ('beesdoo_shift', 'manage_beesdoo_shift_sheet_expected', 'ir.model.access', 508, False)
> ('beesdoo_shift', 'manage_cooperative_status', 'ir.model.access', 509, False)
> ('beesdoo_shift', 'manage_cooperative_exempt_reason', 'ir.model.access', 510, False)
> ('beesdoo_shift', 'read_beesdoo_shift_journal', 'ir.model.access', 511, False)
> ('beesdoo_shift', 'access_cooperative_status', 'ir.model.access', 512, False)
> ('beesdoo_shift', 'read_cooperative_exempt_reason', 'ir.model.access', 513, False)
> ('beesdoo_shift', 'read_cooperative_status_history', 'ir.model.access', 514, False)
> ('beesdoo_shift', 'access_beesdoo_shift_type', 'ir.model.access', 515, False)
> ('beesdoo_shift', 'access_beesdoo_shift_type', 'ir.model.access', 516, False)
> ('beesdoo_shift', 'access_beesdoo_shift_template', 'ir.model.access', 517, False)
>
> procedure followed:
>
>    1. dropdb odoo12
>    2. createdb odoo12
>    3. odoo/odoo-bin -c ./odoo.conf -d odoo12 -i beesdoo_base
>    4. odoo/odoo-bin -c ./odoo.conf -d odoo12 -i beesdoo_shift
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/beescoop/Obeesdoo/pull/133?email_source=notifications&email_token=AAANSHS24NARGB2V5LVBMADRA7GZPA5CNFSM4KMC3YS2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEKS4FTA#issuecomment-581288652>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAANSHS5JKA5TPAZQBPJPQDRA7GZPANCNFSM4KMC3YSQ>
> .
>


-- 
Thibault François

Commit:
[CHG] b_shift : generated shifts name computation

Commit:
[ADD] b_shift : attended_worker_no

Commit:
[REF] b_shift : attendance sheet validation wizard

Commit:
[UPD] b_shift : attendance sheets views

Commit:
[UPD] b_shifts : traductions and terms

Commit:
[UPD] b_shift : attendance sheet form view

Commit:
[MIG] b_shift : adapt to version 12

elouan-lb commented:
Hello @StCyr,

Installation from scratch is fine for me, directly with following command : 

`odoo/odoo-bin -c ./odoo.conf -d odoo12 -i beesdoo_shift`

Have you managed installing it ?

StCyr commented:
> 
> 
> Hello @StCyr,
> 
> Installation from scratch is fine for me, directly with following command :
> 
> `odoo/odoo-bin -c ./odoo.conf -d odoo12 -i beesdoo_shift`
> 
> Have you managed installing it ?

hmmm.. I don't know.

If I do:

1.    dropdb odoo12
2.    createdb odoo12
3.    odoo/odoo-bin -c ./odoo.conf -d odoo12 -i beesdoo_base
4.   odoo/odoo-bin -c ./odoo.conf -d odoo12 -i beesdoo_shift

I get the same error as mentioned above.

But then when I run `odoo/odoo-bin -c ./odoo.conf -d odoo12 -i beesdoo_shift` once again, this error message doesn't appear anymore. 

Maybe it's not critical?

Commit:
[ADD] b_base : security groups for demo users

Commit:
[MIG] b_shift : adapt unite tests

Commit:
[MIG] b_shift : adapt date objects

elouan-lb commented:
> @elouan-lb Why rename default_task_type_id to task_type_default_id which make less sense
> it's always default_FIELD_NAME here you are mixing the default and the field name...

@tfrancoi , I agree with you, unfortunately :

> For a field like default_XXX, execute sets the (global) default value of the field XXX in the model named by default_model to the field’s value.

I will try to find something else.

> Do not use id as variable name, you shadow a built-in function called id
it's not a big deal but it's not a good habit to take

Thank you, will change !

Commit:
[REF] b_shift : task_type_default -> pre_filled_task_type

Commit:
[FIX] b_shift : datetime.combine() method for python 3.4

Commit:
[MIG] b_shift : adapt kanban colors

Commit:
[MIG] b_shift : adapt kanban views

Commit:
[FIX] b_shift : abstract model

SQL requests were made on abstract table.

Commit:
[FIX] b_shift : emails on validation

Commit:
[MIG] b_shift : adapt views to version 12.0

Commit:
[MIG] b_shift : remove menu badges

Not supported since Odoo 11.

Commit:
[ADD] b_shift : wizard for generation of missing attendance sheets

Commit:
[REF] beesdoo_shift : add comment

remytms committed:
[FIX] b_shift: Having two PO file for fr language break installation

tfrancoi committed:
[REF] Refactoring of shift module

Move counter specific and cooperator status specific to a module call beesdoo_worker_status
Remove dependency of beesdoo_base to beesdoo_shift
beesdoo_worker_status depends on beesdoo_base
Move all attendance sheet related code to a new module attendance sheet

Leave empty interface on beesdoo_shift to be implemented in other module to define
specific behavior of the counter and the status of the cooperator

odoo-pda committed:
[FIX] make beesdoo_shift installable
on an empty db.

odoo-pda committed:
[ADD] macavrac_base module.
(+ small cleans on beesdoo_website_shift)

odoo-pda committed:
[FIX] templates and beesdoo_shift,
[ADD] small changes on macavrac_base

remytms committed:
[FIX] b_worker_status

Fix bug reported in 9.0 on the computation of nextcoundown.
Fix missing add_days_delta function.
Fix missing today var.

remytms committed:
[FIX] b_base,b_worker_status: is_worker field not found

remytms committed:
[FIX] b_shift_attendance name (and move demo data)

remytms committed:
[FIX] beesdoo_shift_attendance: Installation

Test does not succeed yet.

remytms committed:
[FIX] b_shift_attendance: Group error for generic user

remytms committed:
[FIX] b_shift_attendance: Tests

remytms committed:
[FIX] b_shift: _get_counter_date_state_change

remytms committed:
[FIX] b_worker_status

remytms committed:
[FIX] b_shift_attendance

remytms committed:
[FIX] b_website_shift: Template errors

remytms committed:
[CHG] b_base: remove default contact not needed

remytms committed:
[IMP] b_base: Put root user in barcode group

remytms committed:
[MOV] b_shift: Configuration in shift instead of attendance sheet

remytms committed:
[FIX] b_worker_status: tests

remytms committed:
[FIX] b_worker_status: Demo data

remytms committed:
[FIX] b_base: Remove worker information that stand in b_shift

remytms committed:
[FIX] b_shift: Holiday wizard

remytms committed:
[FIX] b_shift: Exemption wizard

remytms committed:
[FIX] b_worker_status: working_mode in demo data

remytms committed:
[FIX] b_shift_attendance: wizard to validate sheet
